esp/esdllib/params2xml.cpp

```diff
@@ -24,6 +24,27 @@
 
 void child2xml(IEsdlDefinition *esdl, IEsdlDefObject &child, StringBuffer &flexpath, IProperties *params, StringBuffer &xmlstr, StringBuffer &path, unsigned flags, double ver, int flexpath_setpoint);
 
+void paramsBaseStruct2xml(IEsdlDefinition *esdl, IEsdlDefStruct *est, IProperties *params, StringBuffer &xmlstr, StringBuffer &path, unsigned flags, double ver)
+{
+    if (esdl && est && est->queryProp("base_type"))
+    {
+        StringBuffer flexpath(path);
+        if (flexpath.length())
+            flexpath.append(".");
+        unsigned flexpath_setpoint = flexpath.length();
```

Comment:
I think this flexpath and flexpath_setpoint can be shared by all the bases, and so could have been passed into paramsBaseStruct2xml rather than being recalculated..  and the same values can also be passed into the recursive call to this function.